src/main/java/de/tum/in/www1/artemis/repository/metrics/ExerciseMetricsRepository.java

```diff
@@ -0,0 +1,157 @@
+            FROM Exercise e
+            WHERE e.course.id = :courseId
+            """)
+    Set<ExerciseInformationDTO> findAllExerciseInformationByCourseId(@Param("courseId") long courseId);
+
+    @Query("""
+            SELECT e.id AS key, c AS value
+            FROM Exercise e
+            LEFT JOIN e.categories c
+            """)
+    Set<Entry<Long, String>> findCategoriesByExerciseIds(@Param("exerciseIds") Set<Long> exerciseIds);
+
+    /**
+     * Get the average score for a set of exercises.
+     *
+     * @param exerciseIds the ids of the exercises
+     * @return the average score for each exercise
+     */
+    @Query("""
+            SELECT new de.tum.in.www1.artemis.web.rest.dto.metrics.ScoreDTO(p.exercise.id, AVG(COALESCE(p.lastScore, 0)))
+            FROM ParticipantScore p
+            WHERE p.exercise.id IN :exerciseIds
+            GROUP BY p.exercise.id
+            """)
+    Set<ScoreDTO> findAverageScore(@Param("exerciseIds") Set<Long> exerciseIds);
+
+    /**
+     * Get the score for a user in a set of exercises.
+     *
+     * @param exerciseIds the ids of the exercises
+     * @param userId      the id of the user
+     * @return the score for the user in each exercise
+     */
+    @Query("""
+            SELECT new de.tum.in.www1.artemis.web.rest.dto.metrics.ScoreDTO(s.exercise.id, CAST(COALESCE(s.lastRatedScore, s.lastScore, 0) AS DOUBLE))
+            FROM StudentScore s
+            WHERE s.exercise.id IN :exerciseIds
+                AND s.user.id = :userId
+            """)
+    Set<ScoreDTO> findScore(@Param("exerciseIds") Set<Long> exerciseIds, @Param("userId") long userId);
+
+    /**
+     * Get the latest submission dates for a user in a set of exercises.
+     *
+     * @param exerciseIds the ids of the exercises
+     * @param userId      the id of the user
+     * @return the latest submission dates for the user in the exercises
+     */
+    @Query("""
+            SELECT new de.tum.in.www1.artemis.web.rest.dto.metrics.ResourceTimestampDTO(e.id, s.submissionDate)
+            FROM Submission s
+                LEFT JOIN StudentParticipation p ON s.participation.id = p.id
+                LEFT JOIN p.exercise e
+                LEFT JOIN p.team t
+                LEFT JOIN t.students u
+            WHERE e.id IN :exerciseIds
+                AND s.submissionDate = (
+                    SELECT MAX(s2.submissionDate)
+                    FROM Submission s2
+                    WHERE s2.participation.id = s.participation.id
+                        AND s2.submitted = TRUE
+                )
+                AND (p.student.id = :userId OR u.id = :userId)
+            """)
+    Set<ResourceTimestampDTO> findLatestSubmissionDatesForUser(@Param("exerciseIds") Set<Long> exerciseIds, @Param("userId") long userId);
```

Comment:
The method name does not seem to fit, as it restricts the submissionDates to a set of exercises (defined by their IDs).